distributed/shuffle/_worker_extension.py

```diff
@@ -409,15 +632,15 @@ async def _(extension: ShuffleWorkerExtension, shuffle: ShuffleRun) -> None:
 
     def add_partition(
         self,
-        data: pd.DataFrame,
+        data: Any,
+        input_partition: int | tuple[int, ...],
         shuffle_id: ShuffleId,
-        input_partition: int,
-        npartitions: int,
-        column: str,
+        type: str,
+        **kwargs: Any,
     ) -> int:
-        shuffle = self.get_or_create_shuffle(
-            shuffle_id, empty=data, npartitions=npartitions, column=column
-        )
+        if type == "DataFrameShuffle":
```

Comment:
This is leaky, we should probably handle differently. One approach would be assembling a `ShuffleSpec` object in the actual task that we pass in here, which would also allow us add a dispatch.